Automattic/wp-calypso
# Stepper: Apply extended props to `calypso_stepper_flow_start`

Part of addressing https://github.com/Automattic/wp-calypso/issues/94108

## Proposed Changes

- Adds `STEPPER_TRACKS_EVENT_FLOW_START`
- Implements props extension for `calypso_stepper_flow_start` event
- Some general refactors to `useFlowAnalytics`. Also converted to `useFlowStartTracking`

**DO NOT MERGE** - depended on ongoing discussions in https://github.com/Automattic/wp-calypso/issues/94175 (similar/same problem)

## Why are these changes being made?


* Allow flows to extend the default props tracked by Stepper for all the major/framework events

## Testing Instructions



- test `/setup/onboarding` and `/setup/newsletter`
- ensure tracking of `calypso_stepper_flow_start` when walking through steps in Stepper works like before
- should fire once per flow

## Pre-merge Checklist
